src/client/delphi_epidata.py

```diff
@@ -20,684 +20,737 @@
+        # Check parameters
+        if states is None or dates is None:
+            raise EpidataBadRequestException("`states` and `dates` are both required")
+        # Set up request
+        params = {
+            "endpoint": "covid_hosp",
+            "states": Epidata._list(states),
+            "dates": Epidata._list(dates),
+        }
+        if issues is not None:
+            params["issues"] = Epidata._list(issues)
+        if as_of is not None:
+            params["as_of"] = as_of
+        # Make the API call
+        return Epidata._request(params)
+
+    # Fetch COVID hospitalization data for specific facilities
+    @staticmethod
+    def covid_hosp_facility(hospital_pks, collection_weeks, publication_dates=None):
+        """Fetch COVID hospitalization data for specific facilities."""
+        # Check parameters
+        if hospital_pks is None or collection_weeks is None:
+            raise EpidataBadRequestException(
+                "`hospital_pks` and `collection_weeks` are both required"
+            )
+        # Set up request
+        params = {
+            "source": "covid_hosp_facility",
+            "hospital_pks": Epidata._list(hospital_pks),
+            "collection_weeks": Epidata._list(collection_weeks),
+        }
+        if publication_dates is not None:
+            params["publication_dates"] = Epidata._list(publication_dates)
+        # Make the API call
+        return Epidata._request(params)
+
+    # Lookup COVID hospitalization facility identifiers
+    @staticmethod
+    def covid_hosp_facility_lookup(state=None, ccn=None, city=None, zip=None, fips_code=None):
+        """Lookup COVID hospitalization facility identifiers."""
+        # Set up request
+        params = {"source": "covid_hosp_facility_lookup"}
+        if state is not None:
+            params["state"] = state
+        elif ccn is not None:
+            params["ccn"] = ccn
+        elif city is not None:
+            params["city"] = city
+        elif zip is not None:
+            params["zip"] = zip
+        elif fips_code is not None:
+            params["fips_code"] = fips_code
+        else:
+            raise EpidataBadRequestException(
+                "one of `state`, `ccn`, `city`, `zip`, or `fips_code` is required"
+            )
+        # Make the API call
+        return Epidata._request(params)
+
+    # Fetch Delphi's COVID-19 Nowcast sensors
+    @staticmethod
+    def covidcast_nowcast(
+        data_source,
+        signals,
+        sensor_names,
+        time_type,
+        geo_type,
+        time_values,
+        geo_value,
+        as_of=None,
+        issues=None,
+        lag=None,
+        **kwargs,
+    ):
+        """Fetch Delphi's COVID-19 Nowcast sensors"""
+        # Check parameters
+        if None in (
+            data_source,
+            signals,
+            time_type,
+            geo_type,
+            time_values,
+            geo_value,
+            sensor_names,
+        ):
```

Comment:
nitpick [nonblocking]: Slight preference for having this all one one line. There's a similar `if` condition on line 563 that _is_ on one line. Would be nice to match.